### PR TITLE
fix: add `.prettierignore` to disable auto-formatting issue across all repositories

### LINKED ISSUE #252: add global prettier disable feature
most of new contributors use code formatters like pretteir which make unwanted changes in PR and this issue is with every repo so why don't we disable it globally

### PR DESCRIPTION
**Pull request by AnimeshKumar923**
**Description**

@derberg 
- As discussed [here](https://asyncapi.slack.com/archives/C02FK3YDPCL/p1689606034380449), I'm creating a PR to address the issue faced.
- This file ignores all the auto-formatting used by prettier code extension across different repositories.

cc: @akshatnema @fmvilas @vishvamsinh28 

**Related issue(s)**
Fixes #252

### CODE REVIEW

**Comment by akshatnema** [comment]
@AnimeshKumar923 @derberg Do we have to update any workflow file as well to include this file in updating global workflows?

**Review by derberg — COMMENTED** [review]
the way it is done now means that this file is just for this repo

to have it replicated, you need to add a new job to https://github.com/asyncapi/.github/blob/master/.github/workflows/global-replicator.yml

something like this one: https://github.com/asyncapi/.github/blob/master/.github/workflows/global-replicator.yml#L91-L107

and I think we should not push to all by default, but probably agree on some github repo topic name that repo maintainer can add to get this file, maybe 🤔

**Comment by AnimeshKumar923** [comment]
> and I think we should not push to all by default, but probably agree on some github repo topic name that repo maintainer can add to get this file, maybe thinking

@derberg 
One suggestion from my side is that we announce it on the contributing channel and ask where to add this file. Can we do this?

**Comment by AnimeshKumar923** [comment]
> the way it is done now means that this file is just for this repo
> 
> to have it replicated, you need to add a new job to https://github.com/asyncapi/.github/blob/master/.github/workflows/global-replicator.yml
> 
> something like this one: https://github.com/asyncapi/.github/blob/master/.github/workflows/global-replicator.yml#L91-L107

Hmmm...I'll have to look into that as I'm unfamiliar with YAML files and workflows stuff, but got some experience while working with this: https://github.com/asyncapi/community/pull/805 so give me some time to figure this out... @derberg 
and I'm proud (a little bit) of myself to have my first major contribution here. :blush:

**Comment by derberg** [comment]
maybe there is a way to use GitHub Search to check in in asyncapi org there is any repo that uses prettier

**Comment by AnimeshKumar923** [comment]
> > and I think we should not push to all by default, but probably agree on some github repo topic name that repo maintainer can add to get this file, maybe thinking
> 
> @derberg One suggestion from my side is that we announce it on the contributing channel and ask where to add this file. Can we do this?

Or adding to this, maybe we can start a discussion on this topic on GitHub and let the maintainer join in and discuss. If required, I can open a PR in every repo that requires the `prettierignore` file and referencing this original issue from here?
What do you say? @derberg

**Comment by derberg** [comment]
first we should clarify what repo requires it

there are 2 ways:
- add prettierignore everywhere, excluding repos where prettier is used
- add prettierignore only to repos that need it

this is what we need to put up for discussion

**Comment by vishvamsinh28** [comment]
> first we should clarify what repo requires it
> 
> there are 2 ways:
> 
> * add prettierignore everywhere, excluding repos where prettier is used
> * add prettierignore only to repos that need it
> 
> this is what we need to put up for discussion

I think we should go with 2nd option

**Comment by derberg** [comment]
and this is the easiest and least aggressive way definitely

**Comment by AnimeshKumar923** [comment]
> first we should clarify what repo requires it
> 
> there are 2 ways:
> 
>     * add prettierignore everywhere, excluding repos where prettier is used
> 
>     * add prettierignore only to repos that need it
> 
> 
> this is what we need to put up for discussion

I would go with the 2nd option and ask where to add the `prettierignore` file, which repos require it.

**Comment by chinma-yyy** [comment]
What if we add a condition on the file replicator workflow to check if the repository has any prettier config file? We must check if the repo has a [`.prettierrc.*`](https://prettier.io/docs/en/configuration.html) file. If yes then don't add the file to the repo

**Comment by derberg** [comment]
> Can we define the list of repositories in the workflow file to specify in which repos we want to replicate this file?

we can provide a list of repos, but we also can specify a github topic that one can add to repo, easier than opening a PR to add new repo to the list. Have a look at https://github.com/asyncapi/generator and `get-global-releaserc` topic, which means that this repo gets always the latest file whenever it is updated in `.github` repo. We can have another one, `get-global-prettierignore` and tell people: `hey, add this topic to repo, in a week from now we will add it to `.github` and then you will get it`

**Comment by AnimeshKumar923** [comment]
> maybe there is a way to use GitHub Search to check in in asyncapi org there is any repo that uses prettier

Will look into this...

**Comment by AnimeshKumar923** [comment]
> Have a look at https://github.com/asyncapi/generator and `get-global-releaserc` topic, which means that this repo gets always the latest file whenever it is updated in `.github` repo. We can have another one, `get-global-prettierignore` and tell people: `hey, add this topic to repo, in a week from now we will add it to `.github` and then you will get it`

Hey @derberg can you tell me more about how the topic integrate with providing updates. Any blog/articles on it?

**Comment by derberg** [comment]
@AnimeshKumar923 no blog/article, it is our custom solution.

- we use https://github.com/derberg/manage-files-in-multiple-repositories to replicate files in multiple repos
- one of features is to push files only to repos that have specific topic (more about [gh topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics))

Imagine we push dedicated new configuration for prettier, so it works this way:
- repo maintainer adds required topic to a repo
- repo maintainer asks `.github` repo maintainers to trigger global workflow replication:
    Screenshot 2023-07-25 at 12 52 00
- bot updates whatever is needed to update

**Comment by AnimeshKumar923** [comment]
> 



> @AnimeshKumar923 no blog/article, it is our custom solution.
> 
>     * we use https://github.com/derberg/manage-files-in-multiple-repositories to replicate files in multiple repos
> 
>     * one of features is to push files only to repos that have specific topic (more about [gh topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics))
> 
> 
> Imagine we push dedicated new configuration for prettier, so it works this way:
> 
>     * repo maintainer adds required topic to a repo
> 
>     * repo maintainer asks `.github` repo maintainers to trigger global workflow replication:
> 
>     * bot updates whatever is needed to update

Let me research about it. Will report you later...

**Comment by derberg** [comment]
but what part you want to research? 
someone just needs to ask community in slack, in `tooling` channel, who wants to get this ignoreprettier file, we give people a week to say they want it and that is it

**Comment by vishvamsinh28** [comment]
> 

I'll ask them

**Comment by vishvamsinh28** [comment]
> but what part you want to research? someone just needs to ask community in slack, in `tooling` channel, who wants to get this ignoreprettier file, we give people a week to say they want it and that is it

I'll ask them

**Comment by AnimeshKumar923** [comment]
> but what part you want to research? 

@derberg By research I mean to have a look at the custom method you mentioned. I'll study about it a bit because this is new tot me. I'm having some college work lately, so getting less time to devote here. Sorry about that, but will do the needful from my side.

> someone just needs to ask community in slack, in `tooling` channel, who wants to get this ignoreprettier file, we give people a week to say they want it and that is it.

Hmm...okay. I guess @vishvamsinh28 already did ask it [here](https://asyncapi.slack.com/archives/CQVJXFNQL/p1690382583567669)

**Comment by akshatnema** [comment]
> we can provide a list of repos, but we also can specify a github topic that one can add to repo, easier than opening a PR to add new repo to the list. Have a look at https://github.com/asyncapi/generator and get-global-releaserc topic, which means that this repo gets always the latest file whenever it is updated in .github repo. We can have another one, get-global-prettierignore and tell people: hey, add this topic to repo, in a week from now we will add it to .github and then you will get it

@derberg I'm good with this approach. We just have to ask the maintainers of all repos to add the tag in their repo, if they want this file to be added.

**Comment by derberg** [comment]
@AnimeshKumar923 do you plan to continue with this one?

**Comment by AnimeshKumar923** [comment]
> @AnimeshKumar923 do you plan to continue with this one?

Hey @derberg :wave: 
Thanks for reminding! 
This one slipped from my mind :sweat_smile: Give me few days and I'll let you know about it. 
It's been a while since I looked at this so need a bit of time to grasp this again...

**Comment by AnimeshKumar923** [comment]
Hi @Shurtu-gal :wave: 

Need your help in reviewing this. Can you please take a look at it and provide feedback? Thank you!

**Review by Shurtu-gal — CHANGES_REQUESTED** [review]
@AnimeshKumar923 I couldn't see any `.prettierignore` file which needs to be replicated.

You should add one like in https://github.com/asyncapi/.github/blob/master/.github/workflows/.releaserc

**Comment by AnimeshKumar923** [comment]
> @AnimeshKumar923 I couldn't see any `.prettierignore` file which needs to be replicated.
> 
> You should add one like in https://github.com/asyncapi/.github/blob/master/.github/workflows/.releaserc

Added the file, please review it once again...

**Comment by AnimeshKumar923** [comment]
> > @AnimeshKumar923 I couldn't see any `.prettierignore` file which needs to be replicated.
> > You should add one like in https://github.com/asyncapi/.github/blob/master/.github/workflows/.releaserc
> 
> Added the file, please review it once again... @Shurtu-gal

**Comment by Shurtu-gal** [comment]
Looks fine to me @AnimeshKumar923, but as I am not the CODEOWNER it would be better to ask for approval of others too.

**Comment by AnimeshKumar923** [comment]
> Looks fine to me @AnimeshKumar923, but as I am not the CODEOWNER it would be better to ask for approval of others too.

Thank you so much for the help! Appreciate it. @Shurtu-gal 

Bringing your kind attention to this @derberg

**Review by derberg — COMMENTED** [review]
please update readme with new topic name

**Comment by AnimeshKumar923** [comment]
> please update readme with new topic name

@derberg Did the required change through f5ad226 Please have a look.

> _NOTE: The URL at L18 of README is currently pointing to my personal fork of this repo. Once this PR is merged, I will update the URL accordingly._

File: README.md [hunk 1]
```diff
@@ -13,5 +13,6 @@ Below is a list of workflows with their corresponding topics and descriptions. B
...
-| `get-global-docs-autoupdate` | [update-docs-on-docs-commits](.github/workflows/update-docs-on-docs-commits.yml) | Updates the documentation of a project using `generate:assets` command (if it exists) when the commit starts with `docs:`
\ No newline at end of file
+| `get-global-releaserc` | [.releaserc](.github/workflows/.releaserc) | Fetches release configuration files from a remote repository andUpdates the documentation of a project using `generate:assets` command (if it exists) when the commit starts with `docs makes them available to other workflows 
+| `get-global-docs-autoupdate` | [update-docs-on-docs-commits](.github/workflows/update-docs-on-docs-commits.yml) | Updates the documentation of a project using `generate:assets` command (if it exists) when the commit starts with `docs:`
+| `get-global-prettierignore` | [replicate-prettierignore-file](https://github.com/AnimeshKumar923/.github-asyncapi/blob/242366b618bd85d896386838ffedf1c9a2dc39b5/.github/workflows/global-replicator.yml#L202C3-L218C61) | Replicates `.prettierignore` file in the repository which has this topic.
```
[the remark this answers is not in the document]
Comment (AnimeshKumar923) [reply README.md:18]: > Link can be changed to `./.github/workflows/global-replicator.yml#L202C3-L218C61`

Thanks for the suggestion. Appreciate it!

[the remark this answers is not in the document]
Comment (AnimeshKumar923) [reply README.md]: resolved with 072810c

**Review by derberg — APPROVED** [review]
Thanks!

Would be great if after merging this PR you could drop a message in tooling channel, for maintainers that they can use new topic to get the global prettier to protect from PRs that apply auto formatting

**Comment by AnimeshKumar923** [comment]
> Thanks!
> 
> Would be great if after merging this PR you could drop a message in tooling channel, for maintainers that they can use new topic to get the global prettier to protect from PRs that apply auto formatting 

Yes, will do for sure! 👍

**Pull request merged**